languages/java-cpg/src/main/java/de/jplag/java_cpg/transformation/operations/TransformationUtil.java

```diff
@@ -0,0 +1,363 @@
+package de.jplag.java_cpg.transformation.operations;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import de.fraunhofer.aisec.cpg.graph.Node;
+import de.fraunhofer.aisec.cpg.graph.edge.PropertyEdge;
+import de.fraunhofer.aisec.cpg.graph.statements.expressions.Block;
+import de.fraunhofer.aisec.cpg.graph.statements.expressions.UnaryOperator;
+import de.fraunhofer.aisec.cpg.helpers.SubgraphWalker;
+import de.jplag.java_cpg.transformation.GraphTransformation;
+
+/**
+ * This class is a collection of auxiliary methods related to {@link GraphTransformation}s.
+ */
+public final class TransformationUtil {
+
+    private static final Logger logger = LoggerFactory.getLogger(TransformationUtil.class);
+    private static final DummyNeighbor DUMMY = DummyNeighbor.getInstance();
+
+    private TransformationUtil() {
+        /* should not be instantiated */
+    }
+
+    /**
+     * Gets the {@link SubgraphWalker.Border} of the given node's sub-AST that links to outer nodes via EOG edges.
+     * @param astRoot the root of the sub-AST
+     * @return the EOG {@link SubgraphWalker.Border} of the AST
+     */
+    public static SubgraphWalker.Border getEogBorders(Node astRoot) {
```

Comment:
What does EOG stand for?

Reply:
That is the Evaluation Order Graph, a concept introduced by the authors of the CPG library, where the term is also often used. The EOG is a type of Control Flow Graph, but instead of blocks, it operates on individual Nodes.
For example, in a binary operation expression, the left operand is evaluated first (except if it is an assignment), then the right operand is evaluated, and then the operation itself. An EOG of a binary operation expression would contain a directed edge from the left operand node to the right operand node and from the right operand to the operation node to reflect that. In control statements, the EOG may split.
In the CPG library, the EOG is the basis for complex name analysis, data flow analysis, and other types of analyses.
I use the EOG as the basis for CPG Linearization.